typescript-carousel/main.ts

```diff
@@ -0,0 +1,104 @@
+const images: string[] = [
+  './images/001.png',
+  './images/004.png',
+  './images/007.png',
+  './images/025.png',
+  './images/039.png',
+];
+
+let currNdx: number = 0;
+
+const $displayImage = document.querySelector(
+  '.display-image'
+) as HTMLImageElement;
+if (!$displayImage) throw new Error('.display-image query failed!');
+
+$displayImage.setAttribute('src', images[currNdx]);
+
+const $carouselContainer = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+if (!$carouselContainer) throw new Error('.carousel-container query failed!');
+
+const $leftArrow = document.querySelector('.fa-chevron-left');
+if (!$leftArrow) throw new Error('.fa-chevron-left query failed!');
+const $rightArrow = document.querySelector('.fa-chevron-right');
+if (!$rightArrow) throw new Error('.fa-chevron-right query failed!');
+
+const $circles = document.querySelectorAll('.fa-circle');
+if (!$circles) throw new Error('.fa-circles query failed!');
+console.log($circles);
+
+$carouselContainer.addEventListener('click', function (event: Event) {
+  console.log(event.target);
+  if (event.target === $leftArrow) {
+    if (currNdx >= 1) {
+      clearCircles();
+      setNdxImageAndDot(currNdx - 1);
+    } else if (currNdx === 0) {
+      clearCircles();
+      setNdxImageAndDot(4);
+    }
+  }
+
+  if (event.target === $rightArrow) {
+    if (currNdx <= 3) {
+      clearCircles();
+      setNdxImageAndDot(currNdx + 1);
+    } else if (currNdx === 4) {
+      clearCircles();
+      setNdxImageAndDot(0);
+    }
+  }
```

Comment:
I would make separate functions for move right and move left and attach them to your right and left arrows instead of having them in this function here.